jetty-server/src/main/java/org/eclipse/jetty/server/Dispatcher.java

```diff
@@ -219,8 +219,18 @@ protected void forward(ServletRequest request, ServletResponse response, Dispatc
 
                 _contextHandler.handle(_pathInContext, baseRequest, (HttpServletRequest)request, (HttpServletResponse)response);
 
-                if (!baseRequest.getHttpChannelState().isAsync())
-                    baseRequest.getResponse().softClose();
+                // If we are not async and not closed already, then close via the possibly wrapped response.
+                if (!baseRequest.getHttpChannelState().isAsync() && !baseResponse.getHttpOutput().isClosed())
+                {
+                    try
+                    {
+                        response.getOutputStream().close();
+                    }
+                    catch (IllegalStateException e)
+                    {
+                        response.getWriter().close();
+                    }
+                }
```

Comment:
There remains only one other usage of `softClose()`. Is it really necessary, or `softClose()` can be removed?

Reply:
The concept of a softClose is still needed by sendError - as we need to be able to treat the output as closed after the sendError for the remaining portion of a dispatch, only to reopen to send the actual response in the errorHandler or ERROR dispatch.    It was wrong to apply that concept to dispatch.forward, as there is no reopen semantic needed after a forward, plus there is a wrapping semantic that we broke.